forc-pkg/src/pkg.rs

```diff
@@ -1444,6 +1514,43 @@ pub fn build(
     Ok((compiled, source_map))
 }
 
+/// Compile the entire forc package and return a CompileAstResult.
+pub fn check(
+    plan: &BuildPlan,
+    silent_mode: bool,
+    sway_git_tag: &str,
+) -> anyhow::Result<CompileAstResult> {
+    let conf = &BuildConfig {
+        print_ir: false,
+        print_finalized_asm: false,
+        print_intermediate_asm: false,
+        silent: silent_mode,
+    };
+
+    let mut namespace_map = Default::default();
+    let mut source_map = SourceMap::new();
+    for (i, &node) in plan.compilation_order.iter().enumerate() {
+        let dep_namespace =
+            dependency_namespace(&namespace_map, &plan.graph, &plan.compilation_order, node);
+        let pkg = &plan.graph[node];
+        let path = &plan.path_map[&pkg.id()];
+        let manifest = ManifestFile::from_dir(path, sway_git_tag)?;
+        let (_, maybe_namespace) =
+            compile(pkg, &manifest, conf, dep_namespace.clone(), &mut source_map)?;
```

Comment:
Should this be `compile_to_ast` rather than `compile`? `compile` does the full compilation of the package (including asm gen) - I think the point of `check` should be to avoid this.

Reply:
omg how did I miss that, I was doing this to get the optional namespace back. I've added that functionality to `compile_ast` now which now fixes this.